Surface_mesh_segmentation/include/CGAL/internal/Surface_mesh_segmentation/SDF_calculation.h

```diff
@@ -137,7 +138,20 @@ class SDF_calculation
     centroid_functor(traits.construct_centroid_3_object()),
     use_diagonal(use_diagonal) 
   {
-    tree.insert(faces(mesh).first, faces(mesh).second, mesh, vertex_point_map);
+
+   face_iterator it, end;
+   for(it = faces(mesh).begin(), end = faces(mesh).end(); it!=end; it++)
+   {
+       typename boost::property_traits<VertexPointPmap>::reference a,b,c;
```

Comment:
Neither with a reference (or at least it won't be what you expect). The code need to be rewritten on any case.